mainapp/helpers.py

```diff
@@ -19,6 +22,9 @@
 book_embed_path = os.path.join(settings.STATICFILES_DIRS[0] + '/mainapp/model_files/surprise/book_embedding.npy')
 sim_books_path = os.path.join(settings.STATICFILES_DIRS[0] + '/mainapp/model_files/surprise/sim_books.pickle')
 
+# For Similar-Genre books
+sim_genre_books_path = os.path.join(settings.STATICFILES_DIRS[0] + '/mainapp/model_files/genre_dictionary.pickle')
```

Comment:
Could you explain, Why is this needed? 
The remaining `n` 's genre as discussed should be taken from the user's most common(rated & recommended) genre and return n top books of that genre not present in `recommended & rated`, I think `df_book` already has all what you need

Reply:
The file `genre_dictionary.pickle` contains inverted index of all different genres, sorted in descending order of votes from column `count` in `full_genre.csv`, making the process less time consuming, else the script would have to loop through a lot of books checking for the highest rating and the corresponding genre. If `count` column (genre votes) is not a directly proportional variable to the book_ratings then I can make an inverted index using books.csv such that
```python
genre_index = {genre: [ [rating, book_id], [rating2, book_id2] ] ,
                         genre2: [ [rating, book_id],[rating, book_id],[rating, book_id] ],
                         genre3: [].......}
```
sorted in descending order of rating. Then directly importing it to the function this variable is being used in.

Reply:
Better reuse [this](https://github.com/Praful932/Kitabe/blob/master/mainapp/helpers.py#L98)function, add a parameter `n_books`(default = 16) - For genre page.
Use this function to get total `2*needed_books` (to avoid duplicates)  for any particular genre